core/src/main/java/com/linecorp/armeria/common/stream/NoopSubscriber.java

```diff
@@ -40,11 +52,21 @@ public void onSubscribe(Subscription s) {
     }
 
     @Override
-    public void onNext(T t) {}
+    public void onNext(T item) {
+        requireNonNull(item, "item");
+        if (whenSubscribed.isDone()) {
+            StreamMessageUtil.closeOrAbort(item);
+            return;
+        }
```

Comment:
This is incorrect. `onNext()` will be invoked before `onComplete()` or `onError()`, so this block is basically no-op. To release all items, you have to do the following:

```suggestion
        StreamMessageUtil.closeOrAbort(item);
```